lib/rucio/daemons/replicarecoverer/suspicious_replica_recoverer.py

```diff
@@ -285,48 +287,50 @@ def declare_suspicious_replicas_bad(once=False, younger_than=3, nattempts=10, vo
                                 # Don't keep log files or user files
-                                files_to_be_declared_bad.append(recoverable_replicas[vo][rse_key][replica_key]['surl'])
+                                files_to_be_declared_bad.append(recoverable_replicas[vo][rse_key][replica_key])
                                 del recoverable_replicas[vo][rse_key][replica_key]
                             else:
                                 # Deal with replicas based on their metadata.
-                                file_metadata = get_metadata(recoverable_replicas[vo][rse_key][replica_key]["scope"], recoverable_replicas[vo][rse_key][replica_key]["name"])
                                 if file_metadata["datatype"] is None:  # "None" type has no function "split()"
-                                    files_to_be_ignored.append(recoverable_replicas[vo][rse_key][replica_key]['surl'])
+                                    files_to_be_ignored.append(recoverable_replicas[vo][rse_key][replica_key])
                                     continue
                                 for i in json_data:
                                     if i["datatype"] == file_metadata["datatype"].split("_")[-1]:
                                         action = i["action"]
                                         if action == "ignore":
-                                            files_to_be_ignored.append(recoverable_replicas[vo][rse_key][replica_key]['surl'])
+                                            files_to_be_ignored.append(recoverable_replicas[vo][rse_key][replica_key])
                                         elif action == "declare bad":
-                                            files_to_be_declared_bad.append(recoverable_replicas[vo][rse_key][replica_key]['surl'])
+                                            files_to_be_declared_bad.append(recoverable_replicas[vo][rse_key][replica_key])
                                         else:
                                             logger(logging.WARNING, "RSE: %s, replica name %s, surl %s: Match for the metadata 'datatype' (%s) of replica found in json file, but no match for 'action' (%s)",
                                                    rse_key, replica_key, recoverable_replicas[vo][rse_key][replica_key]['surl'], i["datatype"], i["action"])
                                         break
                                     else:
                                         # If no policy has be set, default to ignoring the file (no action taken).
-                                        files_to_be_ignored.append(recoverable_replicas[vo][rse_key][replica_key]['surl'])
+                                        files_to_be_ignored.append(recoverable_replicas[vo][rse_key][replica_key])
 
                     logger(logging.INFO, '(%s) Remaining replicas (pfns) that will be ignored:', rse_key)
```

Comment:
Do we need this line anymore?

Reply:
Yes, this is so that we can keep track of which files are being ignored by referencing them in the log file.